### DISCUSSION TITLE
Numbers warping when converting TTF to PNG

### DISCUSSION THREAD

**Discussion by brockdonaldson** [ImageMagick/ImageMagick#3633] [Help]
I've been trying to convert a string of numbers from TTF to a PNG however at certain smaller sizes like 22pt some of the numbers warp and look squished in the output. 

Primarily the 3, 5, and 8. Below is an example with the red numbers being typed directly and appearing how they're suppose to. You'll notice the top portion of the numbers are more rounded than the PNG exported ones.

![image](https://user-images.githubusercontent.com/20830059/117217879-2b86f600-add0-11eb-8a19-26970175680a.png)

I'm currently using

```
convert -background none -fill black -font src/AvenirNextLTPro-Medium.ttf -depth 8 -pointsize 22 label:@- PNG32:NumberSequenceTest.png
```

Any help would be greatly appreciated!

**Comment by snibgo** [comment]
Your black digits seem to be created at pointsize 22, then scaled up. If you want them that size, then use a suitable pointsize, eg 88.

**Reply by brockdonaldson** [reply]
The image attached was just used as an example so I could show the difference clearer. I need the output to be at 22pt. 

Showing the pixel preview the black numbers are still squished in relation to the red:

![image](https://user-images.githubusercontent.com/20830059/117221144-b0750e00-add6-11eb-9053-69e4c5caabc9.png)

**Reply by fmw42** [reply]
Please show original size results.  Your font glyphs may not be antialiased. You may need a different font.

**Reply by brockdonaldson** [reply]
This is the output image:

![NumberSequenceTest](https://user-images.githubusercontent.com/20830059/117221828-101fe900-add8-11eb-96c2-591cd0925e9e.png)

**Reply by fmw42** [reply]
Looks fine to me. What is wrong with it? What you see with the enlarging is the antialiasing. If you do not want the gray, but can put up with stair-stepped black, then add +antialias to your command before label: to turn off the anti-aliasing

**Reply by brockdonaldson** [reply]
My bad, I don't think I explained myself properly in my original questions. I wasn't talking about the font looking blurry or pixelated, I understand that is the antialiasing, but more that the shapes of the characters themselves are warped.

Notice how the top of the 3 is shorter than it should be in the regular red version of the font.  

![image](https://user-images.githubusercontent.com/20830059/117222696-11521580-adda-11eb-8980-e4e634eb959f.png)

**Reply by fmw42** [reply]
Not my expertise. But does it happen for other fonts?

**Reply by brockdonaldson** [reply]
No problem 😄  

It only happens to this font at smaller sizes. I've tried selecting a font size that is 5 times larger and and then downscaling by the same ratio and it fixes the issue, but the font becomes a bit blurrier. (I've also tried `-filter point` but it was pixellated)

```
-pointsize 110 -resize 20% -filter box
```

No worries though. It might be an underlying issue with the font file itself.
